File: Common/src/main/java/possible_triangle/skygrid/platform/services/IPlatformHelper.java
package possible_triangle.skygrid.platform.services;

import net.minecraft.resources.ResourceLocation;
import net.minecraft.tags.Tag;
import net.minecraft.world.level.block.Block;

import java.util.Collection;

public interface IPlatformHelper {

    boolean isDevelopmentEnvironment();

    Collection<ResourceLocation> getTags(Block block);

    Block getBarrier();

    Tag.Named<Block> createBlockTag(ResourceLocation id);

}

File: Common/src/main/kotlin/possible_triangle/skygrid/platform/services/IPlatformHelper.kt
package possible_triangle.skygrid.platform.services

import net.minecraft.resources.ResourceLocation
import net.minecraft.tags.Tag
import net.minecraft.world.level.block.Block
import kotlin.properties.ReadOnlyProperty

interface IPlatformHelper {

    val isDevelopmentEnvironment: Boolean

    fun getTags(block: Block): Collection<ResourceLocation>

    fun createBlock(id: String, block: () -> Block): ReadOnlyProperty<Any?, Block>

    fun createBlockTag(id: ResourceLocation): Tag.Named<Block>
}